src/Common/ThreadPool.h

```diff
@@ -256,6 +256,72 @@ class ThreadFromGlobalPool : boost::noncopyable
     }
 };
 
+/// This class is used by ThreadPool only to allocate threads in GlobalThreadPool.
```

Comment:
Uh, that's a lot of unaesthetic copypasta.

As an alternative idea, what about adding a boolean template parameter with default value to class `ThreadFromGlobalPool`, i.e.
```
template <bool propagage_opentelemetry_context = true>
class ThreadFromGlobalPool
{
```
With that, existing user code needs no change and ThreadPool (l. 327) can instantiate `ThreadFromGlobalPool` with "false"?

Reply:
This is a good idea. But since there's an argument added to `ThreadFromGlobalPool`, even if the argument has default value, existing code still needs to be changed from `ThreadFromGlobalPool` to `ThreadFromGlobalPool<>`.

I think following code would work if we want to keep existing reference to  `ThreadFromGlobalPool` unchanged.

```cpp
template <bool propagage_opentelemetry_context = true>
class ThreadFromGlobalPool2
{
...
};

using ThreadFromGlobalPool = ThreadFromGlobalPool2<>;
```

Reply:
You are right and I have no objections to that.
(Perhaps you could name the class `ThreadFromGlobalPoolImpl` instead `ThreadFromGlobalPoolImpl` - but I am fine with either.)

Reply:
`ThreadFromGlobalPool2` is a demo name, but your proposed name is a good one. Naming is always the most difficult thing.